sql/localsql/nipost/nipost.go

```diff
@@ -8,71 +8,65 @@ import (
 )
 
 type Post struct {
-	Nonce   uint32
-	Indices []byte
-	Pow     uint64
+	Nonce     uint32
+	Indices   []byte
+	Pow       uint64
+	Challenge []byte
 
 	NumUnits      uint32
 	CommitmentATX types.ATXID
 	VRFNonce      types.VRFPostIndex
 }
 
-func AddInitialPost(db sql.Executor, nodeID types.NodeID, post Post) error {
+func AddPost(db sql.Executor, nodeID types.NodeID, post Post) error {
 	enc := func(stmt *sql.Statement) {
 		stmt.BindBytes(1, nodeID.Bytes())
 		stmt.BindInt64(2, int64(post.Nonce))
 		stmt.BindBytes(3, post.Indices)
 		stmt.BindInt64(4, int64(post.Pow))
-		stmt.BindInt64(5, int64(post.NumUnits))
-		stmt.BindBytes(6, post.CommitmentATX.Bytes())
-		stmt.BindInt64(7, int64(post.VRFNonce))
+		stmt.BindBytes(5, post.Challenge)
+		stmt.BindInt64(6, int64(post.NumUnits))
+		stmt.BindBytes(7, post.CommitmentATX.Bytes())
+		stmt.BindInt64(8, int64(post.VRFNonce))
 	}
 	if _, err := db.Exec(`
-		insert into initial_post (
-			id, post_nonce, post_indices, post_pow, num_units, commit_atx, vrf_nonce
-		) values (?1, ?2, ?3, ?4, ?5, ?6, ?7);`, enc, nil,
+		REPLACE into post (
+			id, post_nonce, post_indices, post_pow, challenge, num_units, commit_atx, vrf_nonce
+		) values (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8);`, enc, nil,
```

Comment:
Wdym? The point is to overwrite the existing POST proof with a new one.

Reply:
In my opinion this should never be necessary if we just keep what is in `initial_post`:

- A new node will create an initial post that can be kept forever in localDB to be able to create certification requests
- An existing node will import their initial post during startup with a database migration and it will also be kept forever

Even with nodes that have to resync this is quite easy:

1. During the "create initial post" step if no post is in `initial_post`, check if this is the first ATX for an identity
2. If not use any previous post of the identity

This only breaks in one specific case: The node just restored a checkpoint AND the operator decided to delete their `localsql` (which they shouldn't do after a checkpoint). In that specific case we can fall back to creating a new initial post only for the sake of having one for the purpose of requesting a certificate.

Reply:
It's not only this specific case. The user will not find the initial POST after checkpoint if:
- he decided to remove post.bin in the past (which is fine)
- he already ran v1.3 - this version removes the initial post from the DB by itself after publishing an ATX.

The second point is the real problem (the code removing the initial post from DB is already used in prod). 

IMO, there is no downside to keeping the last POST proof in the local DB and it can improve UX of these users (they will avoid regenerating the initial post after checkpoint).